pkgs/instrumentations/web/web-vitals/src/web-vitals.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+import { ReportOpts, Metric } from 'web-vitals';
+import * as webVitals from 'web-vitals';
+import { events } from '@opentelemetry/sandbox-api-events';
+
+import { VERSION } from './version';
+
+const DEFAULT_METRIC_NAMES: Metric['name'][] = ['CLS', 'FID', 'LCP'];
+
+/**
+ * WebVitalsPlugin Config
+ */
+export interface WebVitalsInstrumentationConfig
+  extends InstrumentationConfig,
+    ReportOpts {
+  /**
+   * List of web-vitals to instrument ('CLS', 'FID', 'LCP', 'FCP', 'INP', 'TTFB')
+   * By default 'CLS', 'FID and 'LCP' are instrumented
+   */
+  metrics?: Metric['name'][];
+}
+
+export class WebVitalsInstrumentation extends InstrumentationBase {
+  private _metrics: Metric['name'][];
+  private _reportAllChanges: ReportOpts['reportAllChanges'];
+  private _durationThreshold: ReportOpts['durationThreshold'];
+
+  constructor(config?: WebVitalsInstrumentationConfig) {
+    super('@opentelemetry/sandbox-instrumentation-web-vitals', VERSION, config);
+    this._metrics = config?.metrics || DEFAULT_METRIC_NAMES;
+    this._reportAllChanges = config?.reportAllChanges;
+    this._durationThreshold = config?.durationThreshold;
+  }
+
+  private _onReport(metric: Metric) {
+    const eventEmitter = events.getEventEmitter('web-vitals', VERSION);
+    eventEmitter.emit({
+      name: metric.name,
```

Comment:
We should probably use 'web_vital' here and add metric.name to the event.data

Reply:
Abinet is correct - that's how we defined it [here](https://github.com/open-telemetry/semantic-conventions/pull/30/files#diff-31cea6f5b1661b1a8bba88fd1a4594cb6455dcfd09c969c46db15dba3d718f0fR190)